dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -47,39 +61,71 @@ protected PageResource(HTMLPageAssetAPI pageApi, HostAPI hostApi, WebResource we
-    @GET
-    @JSONP
-    @Path("/page/{pageId}")
     @NoCache
-    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public Map<String, String> list(@Context HttpServletRequest request, @PathParam("pageId") String pageId) {
-
-        Map<String, String> hash = Maps.newHashMap();
-
-
-        return hash;
-    }
-
-    /**
-     * <p>Returns a JSON representation of the Rule with the given ruleId
-     * <p/>
-     * Usage: GET api/rules-engine/sites/{siteId}/rules/{ruleId}
-     */
     @GET
-    @JSONP
-    @Path("/rules/{ruleId}")
-    @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public String self(@Context HttpServletRequest request, @PathParam("siteId") String siteId, @PathParam("ruleId") String ruleId) {
-        Map<String, String> hash = Maps.newHashMap();
-
-
-        return "asd";
+    @Path("/json/{uri: .*}")
+    public Response loadJson(@Context HttpServletRequest request, @Context HttpServletResponse
+            response, @PathParam("uri") String uri) {
+        // Force authentication
+        final InitDataObject auth = webResource.init(false, request, false);
+        final User user = auth.getUser();
+        Response res = null;
+        try {
+            final PageView pageView = this.pageResourceHelper.getPageData(request, response,
+                    user, uri);
+            final String json = this.pageResourceHelper.asJson(pageView);
+            final Response.ResponseBuilder responseBuilder = Response.ok(json);
+            responseBuilder.header("Access-Control-Expose-Headers", "Authorization");
+            responseBuilder.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, " +
+                    "Content-Type, " + "Accept, Authorization");
+            res = responseBuilder.build();
+        } catch (JsonProcessingException e) {
+            res = ExceptionMapperUtil.createResponse(null, "An error occurred when generating " +
+                    "the JSON response (" + e.getMessage() + ")");
+        } catch (DotSecurityException e) {
+            res = ExceptionMapperUtil.createResponse(null, "The user does not have the required"
+                    + " permissions (" + e.getMessage() + ")");
+        } catch (DotDataException e) {
+            res = ExceptionMapperUtil.createResponse(null, "An error occurred when accessing " +
+                    "the page information permissions (" + e.getMessage() + ")");
+        } catch (Exception e) {
+            res = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+        return res;
     }
 
-
-    private User getUser(@Context HttpServletRequest request) {
-        return webResource.init(true, request, true).getUser();
+    @NoCache
+    @GET
+    @Path("/render/{uri: .*}")
+    public Response renderPage(@Context HttpServletRequest request, @Context HttpServletResponse
+            response, @PathParam("uri") String uri) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)